llvm/include/llvm/ADT/APFloat.h

```diff
@@ -951,6 +951,19 @@ class APFloat : public APFloatBase {
 
   bool needsCleanup() const { APFLOAT_DISPATCH_ON_SEMANTICS(needsCleanup()); }
 
+  //  Checks that the value x is in the range (-1;-0.5], [0.5; 1)
+  static bool isInRange(const llvm::APFloat &x) {
```

Comment:
I don't think this should be a member function of `APFloat`. It can be an implementation detail of `ExprConstant.cpp`. In either case, the function should be named something like `isInFrexpResultRange`.